src/obfuscation.ts

```diff
@@ -1,14 +1,77 @@
 import base64 = require('js-base64');
 import * as SparkMD5 from 'spark-md5';
 
+import { PrecomputedFlag } from './interfaces';
+
 export function getMD5Hash(input: string): string {
   return SparkMD5.hash(input);
 }
 
+function saltedHasher(salt: string) {
```

Comment:
minor: taking `Salt` as input would be slightly easier to understand and would make it more obvious what salt representation we're hashing